source/common/stats/thread_local_store.h

```diff
@@ -16,6 +16,142 @@
 namespace Envoy {
 namespace Stats {
 
+/**
+ * Log Linear Histogram implementation per thread.
+ */
+class ThreadLocalHistogramImpl : public Histogram, public MetricImpl {
+public:
+  ThreadLocalHistogramImpl(const std::string& name, Store& parent, std::string&& tag_extracted_name,
+                           std::vector<Tag>&& tags)
+      : MetricImpl(name, std::move(tag_extracted_name), std::move(tags)), parent_(parent),
+        current_active_(0), flags_(0) {
+    histograms_[0] = hist_alloc();
+    histograms_[1] = hist_alloc();
+    cumulative_statistics_.reset(new HistogramStatisticsImpl());
```

Comment:
Same comment as elsewhere. Does this need to be heap allocated? Can't it just be a member directly?